### PR TITLE
🌠signal handling

### PR DESCRIPTION
**Pull request by lucienvb**

### CODE REVIEW

File: src/executor/executor.c [hunk 1]
```diff
@@ -1,9 +1,39 @@
 #include <minishell.h>
 #include "libft.h"
 
+static void	close_pipe(int32_t *pipe_fd)
+{
+	if (close(pipe_fd[READ_END]) != SUCCESS)
+		exit(E_COMMAND_NOT_FOUND);
```
Comment (saladuit) [inline src/executor/executor.c:7]: should call _exit() instead of exit()

Comment (saladuit) [reply src/executor/executor.c:7]: and perhaps print a message

File: src/executor/executor.c [hunk 2]
```diff
@@ -1,9 +1,39 @@
...
+	if (close(pipe_fd[WRITE_END]) != SUCCESS)
+		exit(E_COMMAND_NOT_FOUND);
```
Comment (saladuit) [inline src/executor/executor.c:9]: idem

File: src/minishell/minishell.c [hunk 3]
```diff
@@ -28,21 +46,15 @@ int32_t	minishell_loop(t_minishell *sheldon)
...
-{
-	std_fds[STDIN_FILENO] = dup(STDIN_FILENO);
-	std_fds[STDOUT_FILENO] = dup(STDOUT_FILENO);
-	assert(std_fds[STDIN_FILENO] != ERROR && std_fds[STDOUT_FILENO] != ERROR);
-}
-
-void minishell_init(t_minishell *sheldon, char **envp)
+void	minishell_init(t_minishell *sheldon, char **envp)
 {
+	initialize_signal_handling(&sheldon->status);
```
Comment (saladuit) [inline src/minishell/minishell.c:57]: this should happen after bzero

File: src/parser/parse_heredoc.c [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+int32_t	here_doc(char *delimiter, int fd_write_end)
+{
+	char	*line;
+
+	if (!delimiter || ft_strlen(delimiter) == 0)
+		return (ERROR);
+	while (1)
+	{
+		signal(SIGINT, signal_ctrl_c_heredoc);
```
Comment (saladuit) [inline src/parser/parse_heredoc.c:23]: this one can happen outside of the while loop right? Or should it be reset again after the readline has finished?

**Review by saladuit — CHANGES_REQUESTED** [review]

File: src/executor/executor.c (hunk 1 above)
Comment (lucienvb) [reply src/executor/executor.c:7]: done

File: src/executor/executor.c (hunk 2 above)
Comment (lucienvb) [reply src/executor/executor.c:9]: done

File: src/minishell/minishell.c (hunk 3 above)
Comment (lucienvb) [reply src/minishell/minishell.c:57]: done

File: src/parser/parse_heredoc.c (hunk 4 above)
Comment (lucienvb) [reply src/parser/parse_heredoc.c:23]: you're right, done

**Review by saladuit — APPROVED** [review]
GG WP

**Pull request merged**